zeppelin-zengine/src/test/java/org/apache/zeppelin/interpreter/InterpreterFactoryTest.java

```diff
@@ -17,481 +17,50 @@
+    remoteInterpreter = (RemoteInterpreter) interpreterFactory.getInterpreter("user1", "note1", "double_echo");
+    assertEquals(DoubleEchoInterpreter.class.getName(), remoteInterpreter.getClassName());
   }
 
-  @Test
-  public void testRemoteRepl() throws Exception {
-    interpreterSettingManager = new InterpreterSettingManager(conf, depResolver, new InterpreterOption(true));
-    ArrayList<InterpreterInfo> interpreterInfos = new ArrayList<>();
-    interpreterInfos.add(new InterpreterInfo(MockInterpreter1.class.getName(), "mock1", true, new HashMap<String, Object>()));
-    interpreterSettingManager.add("mock1", interpreterInfos, new ArrayList<Dependency>(), new InterpreterOption(),
-        Maps.<String, DefaultInterpreterProperty>newHashMap(), "mock1", null);
-    Map<String, InterpreterProperty> intp1Properties = new HashMap<String, InterpreterProperty>();
-    intp1Properties.put("PROPERTY_1",
-        new InterpreterProperty("PROPERTY_1", "VALUE_1"));
-    intp1Properties.put("property_2", new InterpreterProperty("property_2", "value_2"));
-    interpreterSettingManager.createNewSetting("mock1", "mock1", new ArrayList<Dependency>(), new InterpreterOption(true), intp1Properties);
-    factory = new InterpreterFactory(conf, null, null, null, depResolver, false, interpreterSettingManager);
-    List<InterpreterSetting> all = interpreterSettingManager.get();
-    InterpreterSetting mock1Setting = null;
-    for (InterpreterSetting setting : all) {
-      if (setting.getName().equals("mock1")) {
-        mock1Setting = setting;
-        break;
-      }
-    }
-    InterpreterGroup interpreterGroup = mock1Setting.getInterpreterGroup("user", "sharedProcess");
-    factory.createInterpretersForNote(mock1Setting, "user", "sharedProcess", "session");
-    // get interpreter
-    assertNotNull("get Interpreter", interpreterGroup.get("session").get(0));
-    assertTrue(interpreterGroup.get("session").get(0) instanceof LazyOpenInterpreter);
-    LazyOpenInterpreter lazyInterpreter = (LazyOpenInterpreter)(interpreterGroup.get("session").get(0));
-    assertTrue(lazyInterpreter.getInnerInterpreter() instanceof RemoteInterpreter);
-    RemoteInterpreter remoteInterpreter = (RemoteInterpreter) lazyInterpreter.getInnerInterpreter();
-    assertEquals("VALUE_1", remoteInterpreter.getEnv().get("PROPERTY_1"));
-    assertEquals("value_2", remoteInterpreter.getProperty("property_2"));
+  @Test(expected = InterpreterException.class)
+  public void testUnknownRepl1() throws IOException {
+    interpreterSettingManager.setInterpreterBinding("user1", "note1", interpreterSettingManager.getSettingIds());
+    interpreterFactory.getInterpreter("user1", "note1", "test.unknown_repl");
   }
 
-  /**
-   * 2 users' interpreters in scoped mode. Each user has one session. Restarting user1's interpreter
-   * won't affect user2's interpreter
-   * @throws Exception
-   */
   @Test
-  public void testRestartInterpreterInScopedMode() throws Exception {
```

Comment:
@zjffdu Could you point new testcase covers this case?
InterpreterSettingManagerTest looks like have related testcase but somehow restart behavior in scoped mode seems changed. try

1. set an interpreter scoped mode per note
2. create two notes and start a paragraph in each note 
3. Restart intepreter from a note in 'interpreter-binding' menu (not in interpreter menu)

Interpreter supposed to be restart for a note only, but it restarts interpreter for both notes.

Reply:
See test here https://github.com/apache/zeppelin/pull/2554/files#diff-6e873dfc1f240d8453d17a98f2ae4d10R180
and 
https://github.com/apache/zeppelin/pull/2554/files#diff-63ebbe353cb86fc2eb25d29e14e7cb18R101

Reply:
Yes, those tests suppose to ensure restart behavior from note page.
However, if I try this by hands, I'm getting different behavior explained above.

Could you take a look?

Reply:
@Leemoonsoo I believe this is the current behavior, not caused by this PR. Not sure why we would close all interpreter when it is `anonymous`. 
https://github.com/apache/zeppelin/blob/master/zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java#L971

Reply:
You're right. I have tested master branch with shiro configured and this branch without shiro.
I also not sure why close all interpreter when it is `anonymous`.

It's up to you change the behavior in this PR or not, but maybe handle it in separate PR for future reference?